### PR TITLE
fix: optimizing PlayerName batching when it is turned off or turned on

### LINKED ISSUE #2713: PlayerName performance drop on disable
When a player moves and the name starts getting transparent until the alpha is 0, we disable the game object, this has quite a big performance drop on scenes where we have a very large hierarchy

- [ ] Avoid disabling the game object, search for a better solution, maybe just deleting [that line](https://github.com/decentraland/unity-renderer/blob/f6583aaee66747b8a0b34162b6a20e322b0d713f/unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/PlayerName.cs#L77) solves this issue.

### PR DESCRIPTION
**Pull request by Antivortex**
## What does this PR change?

This PR removed gameObject.SetActive calls from PlayerName class in methods responsible for showing and hiding the player names because of various reasons. Instead it is using a list of canvas renderers and modifies their alpha which does not have such a huge performance overhead as gameObject SetActive

Closes #2713 

...

## How to test the changes?

Go with two peer avatar characters to the same area and check how the name of one character (over his head) is being shown and hidden from the perspective of other character during interactions like this:

1. Rotating camera when turning away from peer avatar character, then rotating back to him
2. Running away from the peer avatar character until his name disappears
3. Logging out on the peer avatar character side and seeing his name disappear together with him.
4. Teleporting peer avatar character and seeing his name disappear.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by kwspgoliasz — APPROVED** [review]
The issue appears to be implemented correctly. Depending on the scenario, the username displayed above the avatar functions in the following manner:

1. When rotating the camera, the username goes out of the user's view in accordance with the camera angle. The username itself is fully visible at all times without fading out.
2. When moving away from the avatar, the username slowly fades out until disappearing completely.
3. When exiting the game or teleporting, the username disappears along with the avatar with no noticeable lags between the two.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/Tests/PlayerNameShould.cs [hunk 1]
```diff
@@ -65,13 +65,13 @@ public void SetNameCorrectly(string name)
...
     [Test]
-    public void DisableTheGameObjectIfHidden()
+    public void DisableRenderersIfHidden()
     {
         playerName.alpha = 0;
         playerName.targetAlpha = 0;
         playerName.Update(float.MaxValue);
 
-        Assert.IsFalse(playerName.gameObject.activeSelf);
+        Assert.IsTrue(playerName.AllRenderersInvisible());
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/Tests/PlayerNameShould.cs:74]: i dont like the exposure of `AllRenderersInvisible` method just for testing purposes. We are adding production functionality just for testing code. You should be able to check the alpha status of the internal `canvasRenderers` list from this test implementation

**Review by lorux0 — CHANGES_REQUESTED** [review]

**Review by kwspgoliasz — APPROVED** [review]
The feature appears to be implemented correctly.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/PlayerName.cs [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 using System.Collections.Generic;
+using System.Runtime.CompilerServices;
 using Cysharp.Threading.Tasks;
 using DCL;
 using TMPro;
 using UnityEngine;
 using UnityEngine.UI;
 
+
+[assembly: InternalsVisibleTo("PlayerNameTests")]
```
[the remark this answers is not in the document]
Comment (Antivortex) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/PlayerName.cs]: @lorux0 
Turns out AssemblyInfo was already there for this assembly, so removing the reference from PlayerName was enough.

**Review by lorux0 — APPROVED** [review]

**Pull request merged**